lib/src/repositories/ton_wallet_repository/ton_wallet_repository_impl.dart

```diff
@@ -197,8 +197,6 @@ mixin TonWalletRepositoryImpl implements TonWalletRepository {
       unsubscribe(asset.address);
     }
 
-    lastUpdatedAssets = assets;
```

Comment:
If you don't update`lastUpdatedAssets`, then you mustn't use `last != null` expression. Imho, it's better to remove `lastUpdatedAssets`, trying to use `wallets` for all checks. Look through the same behaviour in `TokenWalletrepositoryImpl`